core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -242,6 +242,42 @@ case class PSBT(
     PSBT(globalMap, newInputMaps, outputMaps)
   }
 
+  def addScriptWitnessToInput(
+      scriptWitness: ScriptWitness,
+      index: Int): PSBT = {
+    require(
+      index < inputMaps.size,
+      s"index must be less than the number of input maps present in the psbt, $index >= ${inputMaps.size}")
+    require(!inputMaps(index).isFinalized,
+            s"Cannot update an InputPSBTMap that is finalized, index: $index")
+
+    val previousElements = inputMaps(index).elements
+      .filterNot(_.key.head == PSBTInputKeyId.WitnessScriptKeyId.byte)
+
+    val newElement = scriptWitness match {
+      case p2wpkh: P2WPKHWitnessV0 =>
+        InputPSBTRecord.WitnessScript(P2PKHScriptPubKey(p2wpkh.pubKey))
+      case p2wsh: P2WSHWitnessV0 =>
+        InputPSBTRecord.WitnessScript(p2wsh.redeemScript)
+      case _ =>
```

Comment:
don't use `case _ =>`, it breaks pattern match exhaustiveness. You need to think of all the possible cases here and enumerate them. The compiler will even tell you what case you are missing: 

```
[W]  [E1] core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala:257:22
[W]       match may not be exhaustive.
[W]       It would fail on the following input: EmptyScriptWitness
[W]       L257:     val newElement = scriptWitness match {
[W]                                  ^
[W] core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala: L257 [E1]
Compiled core (3024ms)
Compiling core (30 Scala sources)
[W]  [E3] core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala:257:22
[W]       match may not be exhaustive.
[W]       It would fail on the following input: EmptyScriptWitness
[W]       L257:     val newElement = scriptWitness match {
```

Reply:
I tried to do that but I get this
```
[error] /home/ben/projects/bitcoin-s/core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala:272:15: not found: type EmptyScriptWitness
[error]       case _: EmptyScriptWitness =>
[error]               ^
```

importing it does not fix it either

Reply:
It's because `EmptyScriptWitness` is a case object, rather than a `case class`. You need to do `case EmptyScriptWitness => ...`